meta/include/gnuradio-4.0/meta/UncertainValue.hpp

```diff
@@ -0,0 +1,331 @@
+#ifndef GNURADIO_UNCERTAINVALUE_HPP
+#define GNURADIO_UNCERTAINVALUE_HPP
+
+#include <atomic>
+#include <complex>
+#include <concepts>
+#include <cstdint>
+#include <numbers>
+#include <optional>
+#include <type_traits>
+
+#include <gnuradio-4.0/meta/utils.hpp>
+
+namespace gr::meta {
+
+/**
+ *
+ * @brief Propagation of Uncertainties
+ *
+ * original idea by: Evan Manning, "Uncertainty Propagation in C++", NASA Jet Propulsion Laboratory,
+ * C/C++ Users Journal Volume 14, Number 3, March, 1996
+ * http://www.pennelynn.com/Documents/CUJ/HTML/14.03/MANNING/MANNING.HTM
+ *
+ * implements +,-,*,/ operators for basic arithmetic and complex types, for details see:
+ * https://en.wikipedia.org/wiki/Propagation_of_uncertainty#Example_formulae
+ * This implements only propagation of errors between variables that are uncorrelated.
+ * A more rigorous treatment would require the calculation and propagation of the
+ * corresponding covariance matrix which is out of scope of this implementation.
+ */
+
+template<typename T>
+concept arithmetic_or_complex_like = std::is_arithmetic_v<T> || complex_like<T>;
+
+template<arithmetic_or_complex_like T>
+struct UncertainValue {
+    using value_type = T;
+
+    T value       = static_cast<T>(0); /// mean value
+    T uncertainty = static_cast<T>(0); /// uncorrelated standard deviation
+
+    // Default constructor
+    constexpr UncertainValue() noexcept = default;
+
+    constexpr UncertainValue(T value_, T uncertainty_) noexcept : value(value_), uncertainty(uncertainty_) {}
+
+    explicit(false) constexpr UncertainValue(T value_) noexcept : value(value_), uncertainty(static_cast<T>(0)) {}
+
+    constexpr UncertainValue(const UncertainValue &) noexcept = default;
+    constexpr UncertainValue(UncertainValue &&) noexcept      = default;
+    constexpr UncertainValue &
+    operator=(const UncertainValue &) noexcept
+            = default;
+    constexpr UncertainValue &
+    operator=(UncertainValue &&) noexcept
+            = default;
+    ~UncertainValue() = default;
+
+    constexpr UncertainValue &
+    operator=(const T &other) noexcept {
+        value       = other;
+        uncertainty = static_cast<T>(0);
+        return *this;
+    }
+
+    constexpr UncertainValue &
+    operator=(T &&other) noexcept {
```

Comment:
If this is meant to work only with arithmetic and complex, not sure we need the move operations as copy and move are the same for those.